File: backend/entity/Trips.go
package entity

import (
	"gorm.io/gorm"
)

type Trips struct {
	gorm.Model

	Name  string `binding:"required,min=2,max=100"`         // ต้องมีชื่อทริป อย่างน้อย 2 ตัว
	Types string `binding:"required"`                      // ต้องระบุประเภทของทริป
	Days  int    `binding:"required,gte=1,lte=30"`          // ต้องมีจำนวนวันอย่างน้อย 1 วัน สูงสุด 30 วัน

	Con_id uint        `binding:"required"`               // ต้องส่ง Condition ID
	Con    *Condition  `gorm:"foreignKey:Con_id"`

	Acc_id uint           `binding:"required"`            // ต้องส่ง Accommodation ID
	Acc    *Accommodation `gorm:"foreignKey:Acc_id"`

	ShortestPaths []Shortestpath `gorm:"foreignKey:TripID"`             // ไม่ต้อง binding เพราะเป็น relation
}

File: frontend/src/interfaces/Trips.ts
import type { ShortestpathInterface } from "./Shortestpath";

export interface TripInterface {
  ID?: number;              // ตรงกับ gorm.Model.ID (uint)
  CreatedAt?: string;       // gorm.Model.CreatedAt (string ISO date)
  UpdatedAt?: string;       // gorm.Model.UpdatedAt (string ISO date)
  DeletedAt?: string | null;// gorm.Model.DeletedAt (soft delete)

  Name?: string;            // Name
  Types?: string;           // Types
  Days?: number;            // Days

  Con_id?: number;          // Foreign key Condition ID
  Acc_id?: number;          // Foreign key Accommodation ID


  ShortestPaths?: ShortestpathInterface[]; // Array ของ Shortestpath
}
// Created_at     time.Time
// 	Name string 
// 	Types string 
// 	Day time.Time 

// 	Con_id  uint      
//    	Con    *Condition  `gorm:"foreignKey:Con_id"`

// 	Acc_id  uint      
//    	Acc    *Accommodation  `gorm:"foreignKey:Acc_id"`

// 	Path_id  uint      
//    	Path    *Shortestpath  `gorm:"foreignKey:Path_id"`